### PR TITLE
Increase the MAX_THREADPOOL_SIZE limit to 1024

### PR DESCRIPTION
**Pull request by vmore**
This is the result of discussions from #2296  @bnoordhuis . Instead of completely dropping the threshold which was the original proposal in #2296 I just increased the cap.

### CODE REVIEW

File: docs/src/threadpool.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ operations, as well as getaddrinfo and getnameinfo requests.
 
 Its default size is 4, but it can be changed at startup time by setting the
 ``UV_THREADPOOL_SIZE`` environment variable to any value (the absolute maximum
-is 128).
```
Comment (saghul) [inline docs/src/threadpool.rst]: The dos should mention what version included this change.

**Comment by vmore** [comment]
This is pretty much ready to go I just need someone to merge since I don't have write access.

**Comment by bnoordhuis** [comment]
Thanks, landed in 9a10058e. I'll close out the other PR as well.

**Comment by sam-github** [comment]
When this lands in node.js it's worth calling out as a notable change @nodejs/releasers

**Pull request closed without merging**